scheduler/scheduler.cpp

```diff
@@ -1816,6 +1821,45 @@ static void trigger_exit(int signum)
     signal(signum, trigger_exit);
 }
 
+static void handle_scheduler_announce(const char* buf, const char* netname, bool persistent_clients, struct sockaddr_in broad_addr)
+{
+    /* Another scheduler is announcing it's running, disconnect daemons if it has a better version
+       or the same version but was started earlier. */
+    if (!persistent_clients){
+        uint64_t tmp_time;
+        memcpy(&tmp_time, buf + 4, sizeof(uint64_t));
+        time_t other_time = tmp_time;
+        int other_scheduler_protocol = buf[3];
+        string recv_netname;
+        string local_netname = netname;
+        if (other_scheduler_protocol >= 36)
+        {
+            recv_netname = buf + 5 + sizeof(uint64_t);
```

Comment:
std::string has a constructor that takes a pointer and a count, I believe that is what you want to use.